internal/metrix/meter.go

```diff
@@ -0,0 +1,190 @@
+	count(m.ssh.rekeyed, p, success)
+}
+
+// SSHRenewed implements [authority.Meter] for [Meter].
+func (m *Meter) SSHRenewed(provisioner string, success bool) {
+	count(m.ssh.renewed, provisioner, success)
+}
+
+// SSHSigned implements [authority.Meter] for [Meter].
+func (m *Meter) SSHSigned(provisioner string, success bool) {
+	count(m.ssh.signed, provisioner, success)
+}
+
+// SSHAuthorized implements [authority.Meter] for [Meter].
+func (m *Meter) SSHAuthorized(provisioner string, success bool) {
+	count(m.ssh.authorized, provisioner, success)
+}
+
+// SSHEnriched implements [authority.Meter] for [Meter].
+func (m *Meter) SSHEnriched(provisioner string, success bool) {
+	count(m.ssh.enriched, provisioner, success)
+}
+
+// X509Rekeyed implements [authority.Meter] for [Meter].
+func (m *Meter) X509Rekeyed(provisioner string, success bool) {
+	count(m.x509.rekeyed, provisioner, success)
+}
+
+// X509Renewed implements [authority.Meter] for [Meter].
+func (m *Meter) X509Renewed(provisioner string, success bool) {
+	count(m.x509.renewed, provisioner, success)
+}
+
+// X509Signed implements [authority.Meter] for [Meter].
+func (m *Meter) X509Signed(provisioner string, success bool) {
+	count(m.x509.signed, provisioner, success)
+}
+
+// X509Authorized implements [authority.Meter] for [Meter].
+func (m *Meter) X509Authorized(provisioner string, success bool) {
+	count(m.x509.authorized, provisioner, success)
+}
+
+// X509Enriched implements [authority.Meter] for [Meter].
+func (m *Meter) X509Enriched(provisioner string, success bool) {
+	count(m.x509.enriched, provisioner, success)
+}
+
+func count(cv *prometheus.CounterVec, provisioner string, success bool) {
+	cv.WithLabelValues(provisioner, strconv.FormatBool(success)).Inc()
+}
+
+// KMSSigned implements [authority.Meter] for [Meter].
+func (m *Meter) KMSSigned() {
+	m.kms.signed.Inc()
+}
+
+// KMSErrors implements [authority.Meter] for [Meter].
+func (m *Meter) KMSError() {
+	m.kms.errors.Inc()
+}
+
+// provisioner wraps the counters exported by provisioners.
+type provisioner struct {
+	rekeyed *prometheus.CounterVec
+	renewed *prometheus.CounterVec
+	signed  *prometheus.CounterVec
+
+	authorized *prometheus.CounterVec
+	enriched   *prometheus.CounterVec
+}
+
+func newProvisioner(subsystem string) *provisioner {
+	return &provisioner{
+		rekeyed: newCounterVec(subsystem, "rekeyed_total", "Number of certificates rekeyed",
+			"provisioner",
+			"success",
+		),
+		renewed: newCounterVec(subsystem, "renewed_total", "Number of certificates renewed",
+			"provisioner",
+			"success",
+		),
+		signed: newCounterVec(subsystem, "signed_total", "Number of certificates signed",
+			"provisioner",
+			"success",
+		),
+		authorized: newCounterVec(subsystem, "authorized_total", "Number of authorizing webhooks called",
+			"provisioner",
+			"success",
+		),
+		enriched: newCounterVec(subsystem, "enriched_total", "Number of enriching webhooks called",
+			"provisioner",
+			"success",
+		),
```

Comment:
I would rename those counters. The `authorized_total` especially can be confusing because the CA also has an authorization process, and the webhook authorization usage is very niche. Just prepending `webhook_` would be enough.

Reply:
They've been renamed.